plugins/plugin-docker/che-plugin-docker-machine/src/test/java/org/eclipse/che/plugin/docker/machine/DockerInstanceTest.java

```diff
@@ -0,0 +1,239 @@
+import org.eclipse.che.api.machine.server.exception.MachineException;
+import org.eclipse.che.api.machine.server.model.impl.LimitsImpl;
+import org.eclipse.che.api.machine.server.model.impl.MachineConfigImpl;
+import org.eclipse.che.api.machine.server.model.impl.MachineImpl;
+import org.eclipse.che.api.machine.server.model.impl.MachineSourceImpl;
+import org.eclipse.che.api.machine.server.spi.InstanceKey;
+import org.eclipse.che.plugin.docker.client.DockerConnector;
+import org.eclipse.che.plugin.docker.client.Exec;
+import org.eclipse.che.plugin.docker.client.LogMessage;
+import org.eclipse.che.plugin.docker.client.MessageProcessor;
+import org.eclipse.che.plugin.docker.client.ProgressMonitor;
+import org.eclipse.che.plugin.docker.client.params.CommitParams;
+import org.eclipse.che.plugin.docker.client.params.PushParams;
+import org.eclipse.che.plugin.docker.machine.node.DockerNode;
+import org.mockito.Mock;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyBoolean;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.anyVararg;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Tests for {@link DockerInstance}.
+ *
+ * @author Anton Korneta
+ */
+@Listeners(MockitoTestNGListener.class)
+public class DockerInstanceTest {
+    private static final String        FILE_PATH    = "/tmp";
+    private static final String        CONTAINER    = "container144";
+    private static final String        OWNER        = "owner12";
+    private static final String        IMAGE        = "iamage12";
+    private static final String        MACHINE_ID   = "machine12";
+    private static final String        WORKSPACE_ID = "workspace12";
+    private static final String        NAME         = "suse-jdk";
+    private static final String        TYPE         = "docker";
+    private static final String        REGISTRY     = "registry";
+    private static final String        REPOSITORY   = "eclipse-che";
+    private static final String        TAG          = "latest";
+    private static final MachineStatus STATUS       = MachineStatus.RUNNING;
+
+    @Mock
+    private LogMessage                 logMessageMock;
+    @Mock
+    private Exec                       execMock;
+    @Mock
+    private DockerConnector            dockerConnectorMock;
+    @Mock
+    private DockerInstanceStopDetector dockerInstanceStopDetectorMock;
+
+    private DockerInstance dockerInstance;
+
+    @BeforeMethod
+    public void setUp() throws IOException {
+        dockerInstance = getDockerInstance();
```

Comment:
I forgot also
4. by using only framework "injection", then you can use field injection instead of constructor injection (often Constructor injection was to be able to test the object) . With field injection, when you've class inheritance, adding a new parameter in top level class doesn't imply to change every constructor in subclasses.